### DISCUSSION TITLE
Rewriting route for serverless function projects (i.e. no more `/api`)

### DISCUSSION THREAD

**Discussion by adrianmcli** [vercel/vercel#6730] [Help] [answered]
I've been enjoying writing serverless functions w/ Typescript (a la this [example](https://vercel.com/docs/functions/supported-languages#using-typescript)).

However, all my routes are like:

- `https://api.example.com/api/`
- `https://api.example.com/api/foo`
- `https://api.example.com/api/bar`

Since I don't plan on having a website at the root level, is there an (easy) way for me to remove the need for the `/api` sub-directory in the URL?

Ideally, I'd like the following instead of the above:

- `https://api.example.com/`
- `https://api.example.com/foo`
- `https://api.example.com/bar`

**Comment by styfle** [comment, marked as answer]
You can use `rewrites` to solve this.

- If you're using Next.js, configure [next.config.js](https://nextjs.org/docs/api-reference/next.config.js/rewrites)
- If you're using any other framework, configure [vercel.json](https://vercel.com/docs/cli#project-configuration/rewrites)

It will look like:

```json
{
  "rewrites": [
    { "source": "/:path*", "destination": "/api/:path*" }
  ]
}
```

This is a wildcard that will match every path and rewrite it to the corresponding api directory.

**Reply by tomasweigenast** [reply]
I know this is old, but does this method come with a request overhead of redirecting?

**Reply by styfle** [reply]
It’s not redirecting, it’s rewriting. So there is no additional round trip and therefore no discernible performance difference.
